### ISSUE TITLE
Gen. Docs for OCA Repos

### ISSUE THREAD

**Issue by euroblaze** [euroblaze/voltAIc#10]
Odoo Community Association is a treasure chest of functional modules. Unfortunately most of them are not properly documented, hence they are not understood and used. 

Gen-AI can change that. 
It is possible to automate the A) Generation and B) Publication of the modules. 

- Use RAG, or fine-tune LLAMA2/Mistral, or both
- Read code-repos programmatically and gen content about what their functionality is. 
- publish at simplify-erp.de website.

**Comment by CaoHoangTung** [comment]
- LLamaFactory deployed on server: http://148.251.195.98:7860/
- Initial testing result: For a dataset of 100k text samples, it would take ~9 hours to complete finetuning

UI
![image](https://github.com/euroblaze/voltAIc/assets/22815550/f527e7fc-7c99-42e4-b295-255928d0038f)


Data example
![image](https://github.com/euroblaze/voltAIc/assets/22815550/e48890ae-2e27-467f-9432-74edaffa67af)

Test training with `mistral-7b` on `alpaca_en` dataset
![image](https://github.com/euroblaze/voltAIc/assets/22815550/a94c73d5-a3d6-4009-8e01-269e20580d70)

**Comment by euroblaze** [comment]
@CaoHoangTung 

### Training Set = 1, 10, 100 Odoo module(s)

Approx 1000 lines of Python + Javascript per module.
Keen to gain insights into.. 

- how long did each training take?
- are we receiving higher quality responses? What is the test-setup to benchmark this?

**Comment by CaoHoangTung** [comment]
[12 Apr] 

@euroblaze 
# How to run model from llama-factory in ollama

## Step 1: Get the trained model in huggingface format
Trying, TBU

## Step 2: Convert huggingface checkpoint to gguf format - tested and work with llama2-7b
Use [llama.cpp](https://github.com/ggerganov/llama.cpp/) to convert the llama huggingface checkpoint to GGUF format- 

## Step 3: Run GGUF model in Ollama - tested and work with llama2-7b
Build a custom model using GGUF file, Use the Ollama GGUF support. Ref: [link](https://www.markhneedham.com/blog/2023/10/18/ollama-hugging-face-gguf-models/) 

---

Next steps:
- Configure a small sample dataset from OCA in llama-factory supported format
- Train in llama-factory, get the training insight
- Verify that the model can be run in ollama

After that
- Work on mechanism for benchmarking
- Format the whole data from OCA
- Training & evaluating on full data

**Comment by euroblaze** [comment]
Hi @CaoHoangTung 

About ..

> Next steps:

> Configure a small sample dataset from OCA in llama-factory supported format

Could you please help me understand what "Configure" entails?

> Train in llama-factory, get the training insight

I guess run `python src/train_bash.py`?

> Look into pre-training mode (which may be more suitable)

Let's get rolling on this too. (feels slow this week yet).

Cheers and thanks for feedback.

**Comment by CaoHoangTung** [comment]
> Could you please help me understand what "Configure" entails?
> 
> >

This basically just means that we need to create a dataset based on the OCA modules that follows the format from `llama-factory`



> I guess run `python src/train_bash.py`?
> 
> >

Correct

**Comment by euroblaze** [comment]
@CaoHoangTung 

# README

Today we want to figure out how to feed the new GGUF model a folder of code, and obtain a README.md, printed out to an NGINX web-root folder.

Put the read of the folder of code in a loop, and let it loop to N. Start loop at randomly selected M.

**Comment by euroblaze** [comment]
@CaoHoangTung 

- Mind if I refactor the [odoo-doc-writer](https://github.com/euroblaze/voltAIc/tree/main/odoo-doc-writer) code to remove variables and prompt to yml and .txt files?
- Meanwhile you can focus on the model-training/fine-tuning problem, according to Overall Goal on #9 (reduce training-time, minimise model-size).

Thanks
